File: 2023/UICPC DIV1/UICPC Round #06/solutions & data/plowking/submissions/accepted/plow-zf.cpp
#include <iostream>
#include <algorithm>

using namespace std;

typedef long long ll;

int main() {
    ll n, m, tot = 0;
    cin >> n >> m;

    ll excess = m-(n-1);

    for (ll j = 1, k = 1; j <= m; ++j, ++k) {
        tot += j;
        if (excess > 0) {
            j += min(excess, k-1);
            excess -= k-1;
        }
        else {
            tot += (m*(m+1) - j*(j+1))/2;
            break;
        }
    }

    cout << tot << endl;

    return 0;
}

File: 2023/UICPC DIV1/UICPC Round #06/solutions & data/plowking/submissions/accepted/plow-zf.py
#!/usr/bin/python3

n, m = map(int, input().split())
tot, excess = 0, m-(n-1)

j, k = 1, 1
while j <= m:
    tot += j
    if excess > 0:
        j += min(excess, k-1)
        excess -= k-1
    else:
        tot += (m*(m+1) - j*(j+1))//2
        break
    j += 1
    k += 1

print(tot)
